presto-teradata-functions/src/main/java/com/facebook/presto/teradata/functions/dateformat/DateFormatParser.java

```diff
@@ -18,50 +18,82 @@
 import com.facebook.presto.teradata.functions.DateFormat;
 import org.antlr.v4.runtime.ANTLRInputStream;
 import org.antlr.v4.runtime.Token;
-import org.joda.time.format.DateTimeFormatter;
-import org.joda.time.format.DateTimeFormatterBuilder;
 
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.time.format.SignStyle;
+import java.time.temporal.ChronoField;
 import java.util.List;
 
 import static com.facebook.presto.spi.StandardErrorCode.INVALID_FUNCTION_ARGUMENT;
 
 public class DateFormatParser
 {
+    private enum Mode
+    {
+        // Do not require leading zero for parsing two position date fields (MM, DD, HH, HH24, MI, SS)
+        // E.g. "to_timestamp('1988/4/8 2:3:4','yyyy/mm/dd hh24:mi:ss')"
+        PARSER(1),
+
+        // Add leading zero for formatting single valued two position date fields (MM, DD, HH, HH24, MI, SS)
+        // E.g. "to_char(TIMESTAMP '1988-4-8 2:3:4','yyyy/mm/dd hh24:mi:ss')" evaluates to "1988/04/08 02:03:04"
+        FORMATTER(2);
+
+        final int minTwoPositionFieldWidth;
+
+        Mode(int value)
+        {
+            this.minTwoPositionFieldWidth = value;
+        }
+    }
+
     private DateFormatParser()
     {
     }
 
     public static DateTimeFormatter createDateTimeFormatter(String format)
+    {
+        return createDateTimeFormatterImpl(format, Mode.FORMATTER);
+    }
+
+    public static DateTimeFormatter createDateTimeParser(String format)
+    {
+        return createDateTimeFormatterImpl(format, Mode.PARSER);
+    }
+
+    private static DateTimeFormatter createDateTimeFormatterImpl(String format, Mode mode)
     {
         DateTimeFormatterBuilder builder = new DateTimeFormatterBuilder();
+        boolean formatContainsHourOfAMPM = false;
         for (Token token : tokenize(format)) {
             switch (token.getType()) {
                 case DateFormat.TEXT:
                     builder.appendLiteral(token.getText());
                     break;
                 case DateFormat.DD:
-                    builder.appendDayOfMonth(2);
+                    builder.appendValue(ChronoField.DAY_OF_MONTH, mode.minTwoPositionFieldWidth, 2, SignStyle.NOT_NEGATIVE);
```

Comment:
static import DAY_OF_MONTH and all the various ChronoFields you use here.
static import NOT_NEGATIVE